omod/src/test/java/org/openmrs/module/appui/rest/SessionControllerTest.java

```diff
@@ -0,0 +1,42 @@
+package org.openmrs.module.appui.rest;
+
+
+import static org.junit.Assert.assertTrue;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.junit.Test;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.util.Map;
+
+/**
+ * Created by user on 17.05.16.
+ */
+public class SessionControllerTest extends BaseModuleWebContextSensitiveTest{
+
+
+    @Autowired
+    SessionController sessionController;
+
+    @Autowired
+    Map<String, Object> map;
+
+    @Test
+    public void shouldSetSessionLocation() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest("POST", "/rest/" + RestConstants.VERSION_1 + "/appui/session");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        Object locationUuid = "9356400c-a5a2-4532-8f2b-2361b3446eb8";
+
+        map.put("location", locationUuid);
+        request.addHeader("Content-Type", "text/json");
+        Object res = sessionController.set(map, request,response);
+
+        assertTrue(res.toString().contains("sessionLocation=Xanadu"));
```

Comment:
I think it would be better to parse res Object to Map but i couldn't do it because of getter definitions conflict ( isRetired/getRetired)